### ISSUE TITLE
Don't call fn if obj key or array el are deleted

### ISSUE THREAD

**Issue by junosuarez** [jden/node-nonblocking#3]
``` js
var obj = {foo: 1, bar: 2}
nonblocking.forEach(obj, console.log)
delete obj.foo
```

current output will be

```
undefined
1
```

console.log should not be called on `undefined`

**Comment by junosuarez** [comment]
fixed in https://github.com/jden/node-nonblocking/commit/89cd25b346d84c6e6cbddab86d43d91430b43d65

**Comment by junosuarez** [comment]
published as nonblocking@1.0.3
